Unity-Technologies/Graphics
# [9.x.x][Shader Graph] Fix Custom Function node reorderable list

### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?
Fixing a regression in ReorderableSlotList where replacing slot types stopped working, causing null references when renaming a slot of the same type. 

---
### Testing status

**Manual Tests**: What did you do?
Created new custom function nodes and subgraph output nodes and renamed the slots. Reordered slots, added, and removed slots to ensure no new regressions are introduced. 
Previous custom function nodes maintain consistent behavior. 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics

Any test projects to go with this to help reviewers?

---
### Comments to reviewers
Notes for the reviewers you have assigned.
